### PR TITLE
Automated GitHub Actions workflow

### PR DESCRIPTION
**Pull request by ryanfortner**
based on [my box64 workflow](https://github.com/ryanfortner/box64-debs/pull/4).

you will need to do a few things before merging:
- [x] disable builds on your builder pi to prevent conflicting changes
- [x] add `GPG_PRIVATE_KEY` and `PASSPHRASE` [actions secrets](https://github.com/Itai-Nelken/weekly-box86-debs/settings/secrets/actions)
- [x] enable cron-triggered builds by removing the comment in lines 5 and 6 of `Update-Box86.yml`

to get the gpg private key for the actions secret, simply run the following command: `gpg --armor --export-secret-key itainelken@gmail.com -w0`. Paste that output into the secret. The passphrase should be your GPG secretkey passphrase.

### CODE REVIEW

File: create-deb.sh [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+get-box86-version ver  || error "Failed to get box86 version!"
+get-box86-version commit || error "Failed to get box86 commit!"
+DEBVER="$(echo "$BOX86VER+$(date +"%F" | sed 's/-//g').$BOX86COMMIT")" || error "Failed to set debver variable."
+
+mkdir doc-pak || error "Failed to create doc-pak dir."
+cp ../docs/README.md ./doc-pak || warning "Failed to add readme to docs"
+cp ../docs/CHANGELOG.md ./doc-pak || error "Failed to add changelog to docs"
+cp ../docs/USAGE.md ./doc-pak || error "Failed to add USAGE to docs"
+cp ../LICENSE ./doc-pak || error "Failed to add license to docs"
```
Comment (Itai-Nelken) [inline create-deb.sh:65]:
```suggestion
cp ../LICENSE ./doc-pak || error "Failed to add license to docs"
cp ../docs/X86WINE.md ./doc-pak
```

File: create-deb.sh [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+echo "Box86 lets you run x86 Linux programs (such as games) on non-x86 Linux systems, like ARM (host system needs to be 32bit little-endian)">description-pak || error "Failed to create description-pak."
+echo "#!/bin/bash
+echo 'Restarting systemd-binfmt...'
+systemctl restart systemd-binfmt || true" > postinstall-pak || error "Failed to create postinstall-pak!"
```
Comment (Itai-Nelken) [inline create-deb.sh]:
```suggestion
systemctl restart systemd-binfmt || echo -e \"\e[1;31mFailed to restart systemd-binfmt! box86 won't get called automatically.\nrebooting might help.\e[0m\"" > postinstall-pak || error "Failed to create postinstall-pak!"
```

**Comment by Itai-Nelken** [comment]
Thanks! sorry it took me so much time to get to this.
I want the script to output the same files as the current one, which is a tar compressed deb in the debian/source folder as well as the deb in the debian/pool folder.

File: create-deb.sh [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+mkdir doc-pak || error "Failed to create doc-pak dir."
+cp ../docs/README.md ./doc-pak || warning "Failed to add readme to docs"
+cp ../docs/CHANGELOG.md ./doc-pak || error "Failed to add changelog to docs"
+cp ../docs/USAGE.md ./doc-pak || error "Failed to add USAGE to docs"
+cp ../docs/X86WINE.md ./doc-pak || error "Failed to add X86WINE to docs"
+cp ../LICENSE ./doc-pak || error "Failed to add license to docs"
+echo "Box86 lets you run x86 Linux programs (such as games) on non-x86 Linux systems, like ARM (host system needs to be 32bit little-endian)">description-pak || error "Failed to create description-pak."
+echo "#!/bin/bash
+echo 'Restarting systemd-binfmt...'
+systemctl restart systemd-binfmt || echo -e \"\e[1;31mFailed to restart systemd-binfmt! box86 won't get called automatically.\nrebooting might help.\e[0m\"" > postinstall-pak || error "Failed to create postinstall-pak!"
```
Comment (Itai-Nelken) [inline create-deb.sh]: This is a change I just made to fix #6:
```suggestion
if ! systemctl restart systemd-binfmt ; then
    echo -e \"\e[1;31mFailed to restart systemd-binfmt! box86 won't get called automatically.\nrebooting might help.\e[0m\"
fi" > postinstall-pak || error "Failed to create postinstall-pak!"
```

**Comment by Itai-Nelken** [comment]
I won't have much time the next few weeks, but hopefully after that I can finally merge this!
The only problem currently is that I want to keep the same "output format" (a deb in debian/pool, and a tar.xz in debian/source).

**Comment by ryanfortner** [comment]
> The only problem currently is that I want to keep the same "output format" (a deb in debian/pool, and a tar.xz in debian/source).

Done. The deb is now moved to debian/pool and tar.xz is placed in debian/source. Make sure you follow the steps on the first comment of this PR before merging.

File: .github/workflows/Update-Box86.yml [hunk 4]
```diff
@@ -0,0 +1,69 @@
+name: Update-Box86
+
+# Controls when the workflow will run
+on:
+  #schedule:
+  #   - cron: '0 */6 * * *'
```
Comment (Itai-Nelken) [inline .github/workflows/Update-Box86.yml]:
```suggestion
  schedule:
     - cron: '0 */6 * * *'
```

**Comment by Itai-Nelken** [comment]
I have finally gotten around to enabling this.
Thanks a lot for doing all the hard work!

**Pull request merged**